### ISSUE TITLE
VerticalBarPlot does not update if xData / yData lists change

### ISSUE THREAD

**Issue by thaapasa** [KoalaPlot/koalaplot-core#52]
Hi! And thanks for your Compose Multiplatform graph library, it's been great for rendering graphs on my Android/iOS app!

I have a view where the user can cycle through different periods, and I update the rendered data directly from old data to new data, retaining the same composable renderers. I noticed that while updating data, the graph would just keep on showing the old values. I traced the problem to the file `io.github.koalaplot.core.bar.VerticalBarPlot.kt`, line 113:

```
@Composable
public fun <X, Y, E : VerticalBarPlotEntry<X, Y>> XYGraphScope<X, Y>.VerticalBarPlot(
    data: List<E>,
    modifier: Modifier = Modifier,
    bar: @Composable BarScope.(index: Int) -> Unit,
    barWidth: Float = 0.9f,
    animationSpec: AnimationSpec<Float> = KoalaPlotTheme.animationSpec
) {
    val dataAdapter = remember {
        EntryToGroupedEntryListAdapter(data)
    }

    // Delegate to the GroupedVerticalBarPlot - non-grouped is like grouped but with only 1 group per x-axis position
    GroupedVerticalBarPlot(
    ...
    )
}
```

There, the `dataAdapter` initialization should be changed to `val dataAdapter = remember(data) { ... }` so that it is updated when the underlying data changes. Or at least by duplicating your code structure and changing that fixed the issue for me...

**Comment by gsteckman** [comment]
Fixed in 0.5.4.

**Comment by thaapasa** [comment]
Thank you! I can also confirm that after updating to 0.5.4 the chart works as expected 👍
